jobs/buildhub/inventory_to_records.py

```diff
@@ -0,0 +1,213 @@
+import argparse
+import asyncio
+import async_timeout
+import csv
+import datetime
+import json
+import logging
+import os
+import re
+import sys
+
+import aiohttp
+import backoff
+
+from .utils import (archive_url, is_release_metadata, is_release_filename,
+                    record_from_url, localize_nightly_url, merge_metadata,
+                    ARCHIVE_URL, FILE_EXTENSIONS, DATETIME_FORMAT)
+
+
+BATCH_SIZE = 4
+NB_RETRY_REQUEST = 3
+TIMEOUT_SECONDS = 5 * 60
+
+
+logger = logging.getLogger(__name__)
+
+
+def read_csv(filename):
+    with open(filename, "r") as csvfile:
+        reader = csv.DictReader(csvfile)
+        for row in reader:
+            yield row
+
+
+@backoff.on_exception(backoff.expo,
+                      asyncio.TimeoutError,
+                      max_tries=NB_RETRY_REQUEST)
+async def fetch_json(session, url):
+    headers = {
+        "Accept": "application/json",
+        "User-Agent": "BuildHub;storage-team@mozilla.com"
+    }
+    with async_timeout.timeout(TIMEOUT_SECONDS):
```

Comment:
Can we catch the TimeOut exception there and print the URL that failed on stderr?